numpy/core/tests/test_numeric.py

```diff
@@ -863,6 +864,30 @@ def test_promote_types_endian(self):
         assert_equal(np.promote_types('<m8', '<m8'), np.dtype('m8'))
         assert_equal(np.promote_types('>m8', '>m8'), np.dtype('m8'))
 
+    def test_can_cast_and_promote_usertypes(self):
+        # The rational type defines safe casting for signed integers,
+        # boolean. Rational itself *does* cast safely to double.
+        # (rational does not actually cast to all signed integers, e.g.
+        # int64 can be both long and longlong and it registers only the first)
+        valid_types = ["int8", "int16", "int32", "int64", "bool"]
```

Comment:
You're missing an integer type here
```suggestion
        valid_types = "bhilqp" + "?"
```
or preferably IMO:
```suggestion
        valid_types = [np.byte, np.short, np.intc, np.int_, np.longlong, np.bool_]
```

Reply:
Its intentional, `rational` skips one integer, because it defines them based on bit-width. I had the first version initially, but need to use bit-width names.

Reply:
Apologies, misread the comment. What you have seems fine.